src/worker.cc

```diff
@@ -193,6 +193,9 @@ void add_date_to_locations(Options& options,
       default:
         break;
     }
+  } else if (!options.date_time().empty() && !locations.empty()) {
+    for (auto& loc : locations)
+      loc.set_date_time(options.date_time());
```

Comment:
if a global `date_time` is set but nothing on the locations, then let's set them on the locations.

Reply:
yeah this is strange i guess. the reasons we have a spot for date time on each location is so that for multipoint routes we can propogate the time at the end of each leg over the course of the route. essentially the date time on the location is an output field not an input field. now that we want to do it with matrix i can see you want per location, ie to change it to an input field. we should consider how this works and possible strive a bit to unconfuse it (maybe we deprecate the global one with a warning and ask that you always put it on your first or last location depending on what you want? i dont know didnt think about it much just yet (well since a year or something the last time i spent brain cells on it)

Reply:
Well, I don't think it's so strange once I thought about it a bit more. It holds true for any other action as well: any input location can start at/arrive by a different time. I think that makes perfect sense. That also goes for isochrones & centroid and maybe even route (see [here](https://github.com/valhalla/valhalla/issues/3789) for more review). I think the global `date_time` is still useful to shortcut setting date_time for e.g. `/route`. There the user would have to set a date_time on the first/last location if there was no global date_time object. It just seems easier UX. While it might seem confusing that you can do both, the code currently allows for it as well. This function is only called if there is no time on any location set, so even now users can do that.

Reply:
> so that for multipoint routes we can propogate the time at the end of each leg over the course of the route.

btw, is anything speaking against allowing a time being set on a location e.g. if it's a `break`? not that it's toooo useful, but could be. wouldn't hurt performance AFAICT

Reply:
hm, what I currently fail to see: how do the tests even work.. I can't see how this would allow a `current` time for e.g. matrix. damn, didn't really think about that.. so the date_time.type is important after all. just in the case we want to set it for all locations, not just the first one..

I tried it in a different which fails the map-matcher tests. Let's talk about it tmrw if we have time.